pkg/validator/validator.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/okteto/okteto/pkg/env"
+)
+
+// ErrReservedVariableName is raised when a variable from cmd option has invalid name
+var ErrReservedVariableName = errors.New("reserved variable name. See documentation for more info: https://www.okteto.com/docs/core/credentials/environment-variables/")
+
+// CheckReservedVariablesNameOption returns an error when any of the variable names from command flags are not allowed as input
+func CheckReservedVariablesNameOption(variables []string) error {
+	for _, v := range variables {
+		name, _, ok := strings.Cut(v, "=")
+		if ok && isReservedVariableName(name) {
+			return fmt.Errorf("%s is %w", name, ErrReservedVariableName)
+		}
+	}
+	return nil
+}
+
+// CheckReservedVarName returns an error when any of the variable names from dependency manifest
+func CheckReservedVarName(variables []env.Var) error {
```

Comment:
`CheckReservedVariablesNameOption` and `CheckReservedVarName` is mostly the same. Could we transform `variables []string` in to  `variables []env.Var` in order to have just one function. Otherwise when we change one function we could forget to update the other one.

Reply:
i did not wanted to make any redundant transformations in order to validate from this point. If you think that would be worth it i can do the transformation for then

Reply:
I agree with Tere it's so simple it's not worth in my opinion to change it